<?xml version="1.0"?>
<launch>

    <arg name="start_pt" default= '-7.43, -15.3, 0.8978652, 0'/>
    <arg name="use_rviz" default="true"/>
    <arg name="use_matplotlib" default="false"/>
    <arg name="is_sim" default="true"/>
    <arg name="map_file" default="$(find svea_core)/resources/maps/floor2.yaml" />

    <group if="$(eval is_sim == true)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/resources/rviz/SVEA_floor2.rviz" />
    </group>

    <node pkg="map_server" type="map_server" name="map_server" output="screen" args="$(arg map_file)" />
    <rosparam command="load" file="$(find svea_core)/resources/params/obstacles.yaml" />
    <rosparam command="load" file="$(find svea_core)/resources/params/track.yaml" />

    <node pkg="svea_core" type="floor2_example.py" name="floor2_example" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
    </node>    
    <node pkg="perception" type="road_perception.py" name="road_perception"> </node>
</launch>
